File: csharp/Platform.Collections.Tests/ListTests.cs
using System.Collections.Generic;
using Xunit;
using Platform.Collections.Lists;


namespace Platform.Collections.Tests
{
    /// <summary>
    /// <para>
    /// Represents the list tests.
    /// </para>
    /// <para></para>
    /// </summary>
    public class ListTests
    {
        /// <summary>
        /// <para>
        /// Tests that get element test.
        /// </para>
        /// <para></para>
        /// </summary>
        [Fact]
        public void GetElementTest()
        {
            var nullList = (IList<int>)null;
            Assert.Equal(0, nullList.GetElementOrDefault(1));
            Assert.False(nullList.TryGetElement(1, out int element));
            Assert.Equal(0, element);
            var list = new List<int>() { 1, 2, 3 };
            Assert.Equal(3, list.GetElementOrDefault(2));
            Assert.True(list.TryGetElement(2, out element));
            Assert.Equal(3, element);
            Assert.Equal(0, list.GetElementOrDefault(10));
            Assert.False(list.TryGetElement(10, out element));
            Assert.Equal(0, element);
        }
    }
}

File: cpp/Platform.Collections.Tests/ListTests.cpp
namespace Platform::Collections::Tests
{
    TEST(ListsTests, GetElementTest)
    {
        {
            auto nullList = std::vector<int>{};
            int element;
            ASSERT_EQ(0, Arrays::GetElementOrDefault(nullList, 1));
            ASSERT_FALSE(Arrays::TryGetElement(nullList, 1, element));
            ASSERT_EQ(0, element);
            // Lists and Arrays are Backward Compatible
            ASSERT_EQ(0, Lists::GetElementOrDefault(nullList, 1));
            ASSERT_FALSE(Lists::TryGetElement(nullList, 1, element));
            ASSERT_EQ(0, element);
        }

        {
            auto array = std::vector<int>{1, 2, 3};
            ASSERT_EQ(3, Lists::GetElementOrDefault(array, 2));
            int element;
            ASSERT_TRUE(Lists::TryGetElement(array, 2, element));
            ASSERT_EQ(3, element);
            ASSERT_EQ(0, Lists::GetElementOrDefault(array, 10));
            ASSERT_FALSE(Lists::TryGetElement(array, 10, element));
            ASSERT_EQ(0, element);
        }
    }
}
